src/cascadia/TerminalSettingsModel/Profile.cpp

```diff
@@ -291,6 +291,7 @@ void Profile::LayerJson(const Json::Value& json)
     JsonUtils::GetValueForKey(json, NameKey, _Name);
     JsonUtils::GetValueForKey(json, GuidKey, _Guid);
     JsonUtils::GetValueForKey(json, HiddenKey, _Hidden);
+    JsonUtils::GetValueForKey(json, SourceKey, _Source);

```

Comment:
'scuse me?

Reply:
Removed

Reply:
Why did you remove it? Why had you added it? It seems very important to serialize it, but i’m surprised it wasn’t there before

Don’t just go deleting lines of code because someone commented on them!

Reply:
Oof. I forgot to run the tests anyways. So, this is interesting. I looked through the history of Profile.cpp and it looks like it was always missing (checked the inheritance and winrt-ification PRs). But, we need this so that the following roundtrip test passes:
```
            {
                "name": "Weird Profile",
                "tabColor": null,
                "foreground": null,
                "source": "local"
            }
```

I think we just always relied on `source` being set by the dynamic profile generator, so we never had to care. But now with inheritance and serialization, we need this? Still a bit weird because really the user shouldn't be setting a `source`, Terminal should. Thoughts? Is this even a valid test?

Reply:
it was set by "generateStub", if I recall correctly.

If there is a source, it must be present in the user's settings.

Reply:
because source is part of the primary key we use to match up dynamic profiles.

Reply:
Ok. Then I added it back in.